zk/client_test.go

```diff
@@ -138,10 +138,46 @@ func (s *ZKClientTestSuite) TestRetryUntilConnectedWithSignal() {
 	z.stop()
 }
 
+// TestZkSizeLimit ensures zk write/update requests fail if size exceeds 1MB
+func (s *ZKClientTestSuite) TestZkSizeLimit() {
+	client := s.CreateAndConnectClient()
+	defer client.Disconnect()
+	numMb := 1024 * 1024
+	legalData := make([]byte, numMb-128)
+	legalData2 := make([]byte, numMb-128)
+	legalData2[0] = 1
+	largeData := make([]byte, numMb)
```

Comment:
nit: `oversizeData` might be a better name

Reply:
fixed